spec/rubocop/cop/rspec/dialect_spec.rb

```diff
@@ -0,0 +1,77 @@
+RSpec.describe RuboCop::Cop::RSpec::Dialect, :config do
+  subject(:cop) { described_class.new(config) }
+
+  let(:cop_config) do
+    {
+      'PreferredMethods' => {
+        'context' => 'describe',
+        'expect' => 'expect'
+      }
+    }
+  end
+
+  it 'allows describe blocks' do
+    expect_no_offenses(<<-RUBY)
+      describe 'display name presence' do
+      end
+    RUBY
+  end
+
+  it 'allows calling methods named context in examples' do
+    expect_no_offenses(<<-RUBY)
+      it 'tests common context invocations' do
+        expect(request.context).to be_empty?
+      end
+    RUBY
+  end
+
+  it 'registers an offense for context blocks' do
+    expect_offense(<<-RUBY)
+      context 'display name presence' do
+      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Prefer `describe` over `context`.
+      end
+    RUBY
+
+    expect_correction(<<-RUBY)
+      describe 'display name presence' do
+      end
+    RUBY
+  end
+
+  it 'registers an offense for RSpec.context blocks' do
+    expect_offense(<<-RUBY)
+      RSpec.context 'context' do
+      ^^^^^^^^^^^^^^^^^^^^^^^ Prefer `describe` over `context`.
+        it 'tests common context invocations' do
+          expect(request.context).to be_empty?
```

Comment:
I don't get why this isn't flagged with the cop config 'expect' => 'expect'

Reply:
Good point. 

@gsamokovarov ?

Reply:
Let me check, I think I had this when I experimented with both `on_block` and `on_send` and testing that the auto-correcting code can handle well both node types.

Reply:
[Rubocop::Cop::MethdoPreferrence#preferred_methods](https://github.com/rubocop-hq/rubocop/blob/master/lib/rubocop/cop/mixin/method_preference.rb#L13-L24) filters entries like `{'foo' => 'foo'}`. If I change:

```diff
- 'expect' => 'expect'
+ 'expect' => 'should'
```

I do get an offense (even though, it does not auto-correct properly, but this is expected).